### DISCUSSION TITLE
MetaPhlAn and HUMAnN tutorials

### DISCUSSION THREAD

**Discussion by kescobo** [EcoJulia/Microbiome.jl#84] [Ideas]
@annelle-abatoni @anikaluo I've started a skeleton of metaphlan and humann tutorials [here](https://github.com/BioJulia/BiobakeryUtils.jl/pull/63) (you can see the rendered version [here](http://biojulia.net/BiobakeryUtils.jl/previews/PR63)), with sections and links to the normal bioBakery tutorials that need to be covered.

I'm going to flesh out the section on installation today, and then after I ping you, I'd like you both to go through those sections and note any problems you encounter or things that are unclear. Once that's done, you should both have the resources you need to run all of the functions, and then I'd like to parcel out different sections for you each to write.

**Comment by kescobo** [comment]
Alright, I've written a bunch of additional things about setup. If you can take a look at the PR above and try to follow the "getting started" page, and the first sections of the metaphlan tutorial, that would be really helpful.

**Comment by anikaluo** [comment]
Here are my notes for Getting Started:
- Line 7-10: "This is a package for the julia programming language, designed for working with the bioBakery family of tools for metagenomic analysis of microbial communities. Currently, we support MetaPhlAn and HUMAnN."
    - Should these link to the code repos or the wikis? I feel like you told me somewhere to link to wikis so I just want to make sure.
- Line 89: "`julia --project=~/my_project`"
    - Not sure if this is accurate, but at the moment I don't think it is specified where exactly `my_project` is created, so maybe saying something like the thing below might be better?
    - "`julia --project=<path to project>` if you're in a different working directory (eg. `julia --project=~/my_project` if `my_project` is in the home directory.".
- Line 93: "(eg [`humann_regroup`](@ref) and [`humann_rename`](@ref))"
    - These links lead to 404s. Should they go somewhere at this moment?
- Line 98
    - I don't have a previous installation and I am new to these commands, so I don't think I can offer valid feedback on this part.
- Line 180
    - My output has a lot more than just those two lines upon first installation. Would it be helpful make it clear that the output may not be just that?
    - I exported my terminal as a txt file which I can share if that would be helpful
- Line 189-192, 195-198
    - Shouldn't these lines be in code blocks?
- Line 202-205
    - Shouldn't these lines not be in code blocks?
- Line 202
    - Again I get a lot more output than you do when I run those functions. Would it be helpful make it clear that the output may not be just that?
    - I exported my terminal as a txt file which I can share if that would be helpful
- Line 208: "`ENV["PATH"] = ENV["PATH"] * ":" * Conda.bin_dir(:biobakery)`"
    - I'm not exactly sure how to run this command? I get a "commmand not found error" when I try to run it in REPL?
- Line 219: "[MetaPhlAn tutorial](metaphlan) or [HUMAnN tutorial](humann)"
    - These links also lead to 404s

**Reply by kescobo** [reply]
Thanks! In future, you can make a PR against this branch, by

```
git checkout tutorials
git pull
git checkout -b tutorials-anika
```

Then when you first open the PR, you can select to merge it into `tutorials` instead of into `main`.
Just make whatever edits, even if you're unsure, I can always cherry-pick

**Reply by kescobo** [reply]
Regarding your comment on line 208 - did you do `using Conda` before you tried to run that?

**Reply by kescobo** [reply]
```
julia> ENV["PATH"] = ENV["PATH"] * ":" * Conda.bin_dir(:biobakery)
"/home/kevin/.emacs.d/bin:/home/kevin/.local/bin:/home/kevin/.cargo/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bi
n:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/kevin/.julia/conda/3/envs/biobakery/bin"
```

**Reply by anikaluo** [reply]
Oh got it. I did that before but then I opened a new julia REPL. That works now, sorry!

**Comment by kescobo** [comment]
@anikaluo @annelle-abatoni Alright, [the humann tutorial](url) is done - I added a bunch of additional functions, but haven't written tests for them. Please review and try to run the tutorial, and if you're up for it, write some tests.

Note, commenting on the PR itself as Annelle [did here](https://github.com/BioJulia/BiobakeryUtils.jl/pull/63#discussion_r723707365) is super useful, because it shows up in VSCode:

![image](https://user-images.githubusercontent.com/3502975/136440277-2e1da646-3d24-4117-b1c0-d16355aba08d.png)
